CovertInfiltration/Src/CovertInfiltration/Classes/X2StrategyElement_DefaultActivityChains.uc

```diff
@@ -10,30 +10,408 @@ class X2StrategyElement_DefaultActivityChains extends X2StrategyElement;
+	
+	Template.ChooseFaction = ChooseMetFaction;
+	Template.ChooseRegions = ChooseRandomContactedRegion;
+	Template.SpawnInDeck = true;
+	Template.NumInDeck = 1;
+	Template.DeckReq = IsUFOChainAvailable;
+	
+	Template.Stages.AddItem('Activity_RecoverUFO');
+	Template.Stages.AddItem('Activity_PrepareUFO');
+	Template.Stages.AddItem('Activity_LandedUFO');
+
+	return Template;
+}
+
+static function bool IsUFOChainAvailable(XComGameState NewGameState, optional XComGameState_ActivityChain ChainState)
+{
+	local XComGameState_HeadquartersAlien AlienHQ;
+
+	AlienHQ = XComGameState_HeadquartersAlien(`XCOMHISTORY.GetSingleGameStateObjectForClass(class'XComGameState_HeadquartersAlien'));
+	
+	if (AlienHQ.bHasPlayerBeenIntercepted || AlienHQ.bHasGoldenPathUFOAppeared || AlienHQ.bHasPlayerAvoidedUFO)
+	{
+		return true;
+	}
+
+	return false;
+}
+
+/*
+static function X2DataTemplate CreateHuntChosenTemplate()
+{
+	local X2ActivityChainTemplate Template;
+
+	`CREATE_X2TEMPLATE(class'X2ActivityChainTemplate', Template, 'ActivityChain_HuntChosen');
+	
+	Template.ChooseFaction = ChooseMetFaction; // TODO: spawn immediately when contacting faction
+	Template.ChooseRegions = ChooseRandomContactedRegion; // TODO: choose region that the chosen is at
+	Template.SpawnInDeck = false;
+	
+	Template.Stages.AddItem('Activity_PrepareChosen');
+	Template.Stages.AddItem('Activity_RecoverChosen');
+	Template.Stages.AddItem('Activity_CommanderChosen');
+	//Template.Stages.AddItem('Activity_ChosenBase');
+	// TODO: when done, reveal chosen base
+
+	return Template;
+}
+*/
+static function X2DataTemplate CreateDestroyFacilityTemplate()
+{
+	local X2ActivityChainTemplate Template;
+
+	`CREATE_X2TEMPLATE(class'X2ActivityChainTemplate', Template, 'ActivityChain_DestroyFacility');
+	
+	Template.ChooseFaction = ChooseMetFaction;
+	Template.ChooseRegions = ChooseRandomContactedRegion; // TODO: choose region where a facility is
+	Template.SpawnInDeck = true;
+	Template.NumInDeck = 1;
+	Template.DeckReq = IsFacilityChainAvailable;
+	
+	Template.Stages.AddItem('Activity_PrepareFacility');
+	Template.Stages.AddItem('Activity_FacilityInformant');
+	//Template.Stages.AddItem('Activity_AvatarFacility');
 
 	return Template;
 }
 
-static function StateObjectReference ChooseMetFaction (XComGameState_ActivityChain ChainState)
+static function bool IsFacilityChainAvailable(XComGameState NewGameState, optional XComGameState_ActivityChain ChainState)
+{
+	local XComGameState_FacilityAlien FacilityState;
+	local XComGameState_WorldRegion RegionState;
+
+	foreach NewGameState.IterateByClassType(class'XComGameState_FacilityAlien', FacilityState)
+	{
+		RegionState = XComGameState_WorldRegion(`XCOMHISTORY.GetGameStateForObjectID(FacilityState.Region.ObjectID));
+
+		if (!RegionState.HaveMadeContact())
+		{
+			return true;
+		}
+	}
+
+	return false;
+}
+// TODO: make something to remove this from the deck if these conditions are ever not met
```

Comment:
TODO, but also - why?

Reply:
If a Facility gets blown up but this chain is still in the deck ready to be drawn, it doesn't make much sense.